### ISSUE TITLE
IT fails on empty BED file in anvilprod

### ISSUE THREAD

**Issue by achave11-ucsc** [DataBiosphere/azul#5537]
[IT in `anvilprod`](https://gitlab.prod.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/11673#L585) expects a non-zero size file. Here's the [file](https://service.prod.anvil.gi.ucsc.edu/index/files/da1608d3-766b-495c-bc6a-34b3514fb83d?catalog=anvil2) of byte size 0, that caused the IT failure.

- [x] Security design review completed; the Resolution of this issue does **not** …
  - [x] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with 
  - [x] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [x] … affect the permissions of external users like access to
    - TDR snapshots
  - [x] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [x] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [x] … affect monitoring of the system
  - [x] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [x] … add an interface that exposes sensitive or confidential data at the security boundary
  - [x] … affect the encryption of data at rest
  - [x] … require persistence of sensitive or confidential data that might require encryption at rest
  - [x] … require unencrypted transmission of data within the security boundary
  - [x] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [x] Documentation on any unchecked boxes is provided in comments below

**Comment by achave11-ucsc** [comment]
Assignee to file partial PR w/ following quick fix.
```diff
Subject: [PATCH] TODO;
---
Index: test/integration_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/integration_test.py b/test/integration_test.py
--- a/test/integration_test.py	(revision 45ba52155523bee39e2b349b50018e43d8b341cb)
+++ b/test/integration_test.py	(date 1694807790380)
@@ -854,7 +854,10 @@
             with gzip.open(content) as buf:
                 self._validate_fastq_content(buf)
         else:
-            self.assertEqual(1, len(content.read(1)))
+            # FIXME: Re-enable assertion
+            #        https://github.com/DataBiosphere/azul/issues/5537
+            if False:
+                self.assertEqual(1, len(content.read(1)))
 
     def _validate_file_response(self, response: urllib3.HTTPResponse, file_ext: str):
         """

```

**Comment by achave11-ucsc** [comment]
@hannes-ucsc: "The long-term solution is to use `within` filter to look for files between 1 and infinity (a very large integer) in size (still sorted by size accendingly) and only if that fails, extend the range to 0 and infinity (will only return empty files). If an empty file was picked the failing assertion should be disabled. If no files were found with either filter the test should still fail."

**Comment by achave11-ucsc** [comment]
Assignee to implement resolution outlined above as a low priority work to fallback to (hobby).

**Comment by nadove-ucsc** [comment]
The snapshot containing the empty file has been removed. https://github.com/DataBiosphere/azul/issues/5690
However the file itself appears to still be present on `anvilprod`: https://service.prod.anvil.gi.ucsc.edu/index/files/da1608d3-766b-495c-bc6a-34b3514fb83d?catalog=anvil2

**Comment by dsotirho-ucsc** [comment]
Assignee requested more time to think.

**Comment by nadove-ucsc** [comment]
The file is no longer present on `anvilprod`.

**Comment by dsotirho-ucsc** [comment]
Assignee to elaborate.

**Comment by nadove-ucsc** [comment]
I believe this issue only manifested in the first place because we indexed [a dataset containing a large number of trash files](https://github.com/DataBiosphere/azul/issues/5690). That dataset, along with all of its trash files, has since been removed. As a result, the symptom is no longer observable. 

It might hypothetically recur if we index another dataset that contains these empty files. Depending on how likely that is, this issue should either be closed or parked. It can't move forward at the moment without a way to verify the fix.
